numcodecs/categorize.py

```diff
@@ -53,7 +53,10 @@ def __init__(self, labels, dtype, astype='u1'):
     def encode(self, buf):
 
         # view input as ndarray
-        arr = ndarray_from_buffer(buf, self.dtype)
+        if self.dtype == object:
+            arr = np.asanyarray(buf, dtype=object).reshape(-1, order='A')
```

Comment:
Just out of curiosity, why is `asanyarray` needed?

Reply:
I'm guessing that any of ``np.array(buf, dtype=object, copy=False``, ``np.asarray(buf, dtype=object)`` or ``np.asanyarray(buf, dtype=object)`` would have the desired effect of creating an array from buf, without copying if buf is already a numpy array with object dtype. Not sure which should be preferred.

Reply:
It's a fair question. My thinking would be one of the first two. `np.asanyarray` seems to suggest we will preserve other `ndarray` subclasses that might include masks, axis tags, or other user customizations. When in reality, we are only serializing the array itself.